SU2_CFD/include/output/COutput.hpp

```diff
@@ -0,0 +1,777 @@
+  PrintingToolbox::CTablePrinter* multiZoneHeaderTable; //!< Multizone header output structure
+  PrintingToolbox::CTablePrinter* historyFileTable;     //!< Table structure for writing to history file
+  PrintingToolbox::CTablePrinter* fileWritingTable;     //!< File writing header
+  std::string multiZoneHeaderString;                    //!< Multizone header string
+  bool headerNeeded;                                    //!< Boolean that stores whether a screen header is needed
+  
+  //! Structure to store the value of the running averages
+  map<string, Signal_Processing::RunningAverage> runningAverages; 
+  
+  //! Structure to store the value initial residuals for relative residual computation
+  std::map<string, su2double> initialResiduals;
+  
+   /*----------------------------- Volume output ----------------------------*/     
+   
+   CParallelDataSorter* volumeDataSorter;    //!< Volume data sorter
+   CParallelDataSorter* surfaceDataSorter;   //!< Surface data sorter
+   
+   vector<string> volumeFieldNames;     //!< Vector containing the volume field names
+   unsigned short nVolumeFields;        /*!< \brief Number of fields in the volume output */ 
+  
+   string volumeFilename,               //!< Volume output filename
+   surfaceFilename,                     //!< Surface output filename
+   restartFilename;                     //!< Restart output filename
+   
+  /** \brief Structure to store information for a volume output field.
+   * 
+   *  The stored information is used to create the volume solution file.   
+   */
+  struct VolumeOutputField {
+    /*! \brief The name of the field, i.e. the name that is printed in the file header.*/
+    string fieldName;
+    /*! \brief This value identifies the position of the values of this field at each node in the ::Local_Data array. */
+    int    offset; 
+    /*! \brief The group this field belongs to. */
+    string outputGroup;
+    /*! \brief String containing the description of the field */
+    string description;
+    /*! \brief Default constructor. */
+    VolumeOutputField () {}
+    /*! \brief Constructor to initialize all members. */    
+    VolumeOutputField(string fieldName_, int offset_, string volumeOutputGroup_, string description_):
+      fieldName(std::move(fieldName_)), offset(std::move(offset_)), 
+      outputGroup(std::move(volumeOutputGroup_)), description(std::move(description_)){}
+  };
+  
+  /*! \brief Associative map to access data stored in the volume output fields by a string identifier. */  
+  std::map<string, VolumeOutputField >          volumeOutput_Map;
+  /*! \brief Vector that contains the keys of the ::volumeOutput_Map in the order of their insertion. */  
+  std::vector<string>                           volumeOutput_List;
+  
+  /*! \brief Vector to cache the positions of the field in the data array */
+  std::vector<short>                            fieldIndexCache;
```

Comment:
if I got this right these variables are associated with `offset`, maybe offset should also be a `short`?